scripts/generate-xN-transpose.sh

```diff
@@ -96,6 +96,13 @@ tools/xngen src/x32-transposec/avx2.c.in -D SIZE=8  IN_PTRS=REUSE OUT_PTRS=SWITC
 tools/xngen src/x32-transposec/avx2.c.in -D SIZE=16 IN_PTRS=REUSE OUT_PTRS=MOV    -o src/x16-transposec/gen/x16-transposec-16x16-reuse-mov-avx2.c &
 tools/xngen src/x32-transposec/avx2.c.in -D SIZE=16 IN_PTRS=REUSE OUT_PTRS=SWITCH -o src/x16-transposec/gen/x16-transposec-16x16-reuse-switch-avx2.c &
 
+#################################### RISCV VECTOR ###############################
+tools/xngen src/x32-transposec/rvv.c.in -D SIZE=32 VLEN=1024 -o src/x32-transposec/gen/x32-transposec-32x8-rvv.c &
```

Comment:
VLEN of 1024 wont be supported on all hardware?

Reply:
In scalar kernels the parameters are
-D SIZE=16 TYPE=int16_t TILE_HEIGHT=2 TILE_WIDTH=1
which allows for different types (int vs float) and any tile size.  Would it make sense to compute VLEN from width/height?

Reply:
X390 supports VLEN 1024. Qemu also supports 1024 vlen, it has been tested. We compute tileH/W from the argument VLEN, not from size. ```$TILE_HEIGHT = int(VLEN/SIZE) ``` This can generate a full vector and an efficient tile size for different HW configs.
I think it makes sense since ARM and other vector code do the same thing.
```
$assert SIZE in [8, 16, 32, 64]
$assert VECTOR_SIZE in [64, 128]
$TILE_SIZE = int(VECTOR_SIZE/SIZE)
```
https://github.com/google/XNNPACK/blob/1bba28d76bbd42c96b58c17644d22e3a85b9e809/src/x32-transposec/neon-zip.c.in#L10

Reply:
X390 supports VLEN 1024. Qemu also supports 1024 vlen and it has been tested. We compute tileH/W using the argument VLEN.` $TILE_HEIGHT = int(VLEN/SIZE) ` This can generate a full vector and an efficient tile size for different HW configs. I think it makes sense since ARM and other vector code do the same thing.
https://github.com/google/XNNPACK/blob/master/src/x32-transposec/neon-zip.c.in#L8-L10